leader-election-impl/src/main/java/com/palantir/leader/proxy/AwaitingLeadershipProxy.java

```diff
@@ -84,6 +87,10 @@ private AwaitingLeadershipProxy(Supplier<?> delegateSupplier,
         this.delegateRef = new AtomicReference<Object>();
         this.interfaceClass = interfaceClass;
         this.isClosed = false;
+
+        String logTag = "Proxy #" + numProxies.getAndIncrement() + " - " + interfaceClass.getSimpleName();
+        log = new TaggedLogger(AwaitingLeadershipProxy.class, logTag);
+        leaderLog = new TaggedLogger("leadership", logTag);
```

Comment:
If we just use different loggers, they can still go to same output (default), but if one desired they could have different appenders (thus different files) if desired.